### PR TITLE
604 project model url

### PR DESCRIPTION
**Pull request by Alastair-L**
### Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/604
### Changes:
One decision made here: 
Deep links beyond just /covidau also redirect to the model, but once you click 'view project' you'll go to the place specified by the link.
e.g.
`/covidau/AU_Victoria`
=> Project model
=> Victoria (once you click 'view project')

### CODE REVIEW

File: packages/web-frontend/src/historyNavigation/historyMiddleware.js [hunk 1]
```diff
@@ -62,14 +63,17 @@ const reactToLocationChange = (store, location, previousLocation) => {
...
     const component = otherComponents[componentKey];
     if (component && component !== previousComponents[componentKey])
       dispatch(setComponent(component));
   };
 
+  if (PROJECTS_WITH_LANDING_PAGES[otherComponents[URL_COMPONENTS.PROJECT]]) {
+    dispatch(setOverlayComponent(PROJECT_LANDING));
+  } else {
+    dispatch(setOverlayComponent(null));
+  }
```
Comment (kael89) [inline packages/web-frontend/src/historyNavigation/historyMiddleware.js:76]: A detail, but I think this makes easier to see what is the common logic:

```suggestion
  const isLandingPageProject = PROJECTS_WITH_LANDING_PAGES[otherComponents[URL_COMPONENTS.PROJECT]];
  const overlayComponent = isLandingPageProject ? PROJECT_LANDING : null;
  dispatch(setOverlayComponent(overlayComponent));
```

**Review by kael89 — APPROVED** [review]
That's a great change! Approving since my comment is an optional suggestion

**Comment by Alastair-L** [comment]
It feels strange having to get stuff tested after all the url refactor cards.

**Comment by kael89** [comment]
> It feels strange having to get stuff tested after all the url refactor cards.

I guess we could include this in the testing scope?

**Comment by Alastair-L** [comment]
I mentioned that here: https://github.com/beyondessential/tupaia-backlog/issues/604#issuecomment-686231719. It's really up to the testers I think.

**Comment by kael89** [comment]
> I mentioned that here: [beyondessential/tupaia-backlog#604 (comment)](https://github.com/beyondessential/tupaia-backlog/issues/604#issuecomment-686231719). It's really up to the testers I think.

Nice, good call!

**Comment by Alastair-L** [comment]
@kael89 could you please take a quick look at this branch, just double check everything is ok?

Since you reviewed:
- Dev was merged into this branch
- To reverse that, I created a new branch based off `url-refactor`, cherry-picked the 3 commits from this branch, and then force pushed
- Then, I created a new commit to fix a bug (which should have been created in the `url-refactor` branch)

File: packages/web-frontend/src/sagas.js [hunk 2]
```diff
@@ -921,6 +922,7 @@ function* exportChart(action) {
     [URL_COMPONENTS.PROJECT]: projectCode,
     [URL_COMPONENTS.ORG_UNIT_NAME]: organisationUnitName,
   });
+  console.log(exportUrl);
```
Comment (kael89) [inline packages/web-frontend/src/sagas.js:925]:
```suggestion
```

**Review by kael89 — APPROVED** [review]
Just one rogue `console.log`, pre-approving

File: packages/web-frontend/src/sagas.js [hunk 3]
```diff
@@ -842,8 +843,10 @@ function* fetchMeasures(action) {
   const projectCode = selectCurrentProjectCode(state);
   const requestResourceUrl = `measures?organisationUnitCode=${organisationUnitCode}&projectCode=${projectCode}`;
   try {
-    const measures = yield call(request, requestResourceUrl);
-    yield put(fetchMeasuresSuccess(measures));
+    const response = yield call(request, requestResourceUrl);
+
+    if (response.measures.length === 0) yield put(clearMeasure());
+    yield put(fetchMeasuresSuccess(response));
```
Comment (Alastair-L) [inline packages/web-frontend/src/sagas.js:848]: Could you please review this @kael89?

**Review by kael89 — APPROVED** [review]

**Pull request merged**